src/resources/measurements.ts

```diff
@@ -1,6 +1,57 @@
 import { Measurement, MeasurementsApi } from '../codegen/api';
 import { Configuration as APIConfiguration } from '../codegen/configuration';
-import { fixMeasurementFields } from '../utils';
+
+interface MeasurementInput extends Omit<Measurement, 'time'> {
+  time?: Date | string;
+}
+
+/**
+ * This helper augments the body with "meter_name" based on the meterName field.
+ *
+ * TODO: API expects "meter_name", while the generated type uses meterName.
+ * This function should be deleted once the API
+ *
+ * @param param0
+ * @returns
+ */
+function fixMeasurementFields({
+  meterName,
+  ...rest
+}: Measurement): Measurement {
+  return (
+    meterName
+      ? {
+          meter_name: meterName,
+          ...rest,
+        }
+      : {
+          ...rest,
+        }
+  ) as Measurement;
```

Comment:
does it work if we just do:
```
const fixedMeasurement: Measurement = meterName ? { ... } : { ... };
return fixedMeasurement;
```

I think the `as` keyword is basically a bail-out AFAIK, so trying to use it less frequently where possible.

Reply:
I do too ([can we lint against it?](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-assertions.md)), but because `meter_name` isn't a property on Measurement, TS rejects the assignment.